IPython/html/files/handlers.py

```diff
@@ -28,12 +28,12 @@ def get(self, path):
         else:
             name = path
         
-        model = cm.get(path)
+        model = cm.get(path, type='file')
         
         if self.get_argument("download", False):
             self.set_header('Content-Disposition','attachment; filename="%s"' % name)
         
-        if model['type'] == 'notebook':
+        if model['type'] == 'notebook' or name.endswith('.ipynb'):
```

Comment:
The few lines below will attempt to guess the mime-type from the extension. This is essentially just ensuring that `.ipynb` is known to be a JSON file, since that's going to be a common case. From the recipient's point of view, it's not just text.

Though possibly we should consider using the mimetype we picked for notebooks, `application/x-ipynb+json`.